### PR TITLE
dynparquet: Allow serializing/deserializing schemas with json

### PR DESCRIPTION
**Pull request by brancz**
@metalmatze @thorfour

### CODE REVIEW

File: dynparquet/schema/schema.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+type SortingColumn struct {
+	Name       string `json:"name"`
+	Order      string `json:"ordering"`
+	NullsFirst bool   `json:"nulls_first"`
+}
+
+func StorageLayoutToParquetNode(l StorageLayout) (parquet.Node, error) {
+	var node parquet.Node
+	switch l.Type {
+	case "string":
```
Comment (metalmatze) [inline dynparquet/schema/schema.go:39]: Can we reuse some sort of constants from other packages for these?

**Review by metalmatze — COMMENTED** [review]
One question about using constants rather than strings, otherwise this is looking great!

File: dynparquet/schema/schema.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+		}
+		node = parquet.Compressed(node, comp)
+	}
+
+	return node, nil
+}
+
+func encodingFromString(enc string) (encoding.Encoding, error) {
+	switch enc {
+	case "RLE_DICTIONARY":
```
Comment (thorfour) [inline dynparquet/schema/schema.go:74]: We have a lot of magic strings in this, 
RLE_DICTIONARY
ascending
descending
and then the types. Can we move all of these to a consts file?

**Comment by brancz** [comment]
I think I've actually decided to go a different route and instead make this protobuf and it ends up being more intertwined with other work than I initially expected. Closing for now.

**Pull request closed without merging**